src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md

```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
 #lang rzk-1
 ```
 
-!!! warning
+Derive the recursion principle for products
+$rec_{A \times B}$ using only the projections,
+and verify that the definitional equalities are valid.
+Do the same for $\Sigma$-types.
 
-    To be done.
+## Solution
+
+### Product $A \times B$.
+
+Introducing definitions
+```rzk
+-- defined before
+
+-- #def id
+--   (A : U)
+--   : A -> A
+--   := \ a -> a
+
+-- #def prod
+--   (A B : U)
+--   : U
+--   := Sigma (_ : A), B
+
+#def pair
+  (A B : U)
+  : A -> B -> prod A B
+  := \ a b -> (a, b)
+```
+
+Defining recursion on $A \times B$ through projections
+```rzk
+#def pr_1_prod
+  (A B : U)
+  : prod A B -> A
+  := \(a, b) -> a
+
+#def pr_2_prod
+  (A B : U)
+  : prod A B -> B
+  := \(a, b) -> b
+
+-- rec_AxB((C: U), (g: A -> B -> C), (p: AxB)) 
+#def rec_prod
+  (A B : U)
+  (C : U)
+  : (g : A -> B -> C) -> (p : prod A B) -> C
+  := \ g p -> g (pr_1_prod A B p) (pr_2_prod A B p)
+```
+
+Defining projections through a recursion on $A \times B$
+```rzk
+-- other way
+#def rec_prod_reversed
+  (A B : U)
+  (C : U)
+  : (g : A -> B -> C) -> (p : prod A B) -> C
+  := \ g (a, b) -> g a b
+
+#def pr_1_prod_reversed
+  (A B : U)
+  : prod A B -> A
+  := rec_prod_reversed A B A (\a b -> a)
+
+#def pr_2_prod_reversed
+  (A B : U)
+  : prod A B -> B
+  := rec_prod_reversed A B B (\a b -> b)
```

Comment:
This is [defined as well](https://rzk-lang.github.io/hottbook/1-foundations/1-type-theory/05-product-types.rzk/#define:pr₁').